csharp/GildedRose.cs

```diff
@@ -1,89 +1,123 @@
-                        }
-                    }
-                }
-                else
+                if (LegendaryItems.Contains(item.Name))
                 {
-                    if (Items[i].Quality < 50)
-                    {
-                        Items[i].Quality = Items[i].Quality + 1;
-
-                        if (Items[i].Name == "Backstage passes to a TAFKAL80ETC concert")
-                        {
-                            if (Items[i].SellIn < 11)
-                            {
-                                if (Items[i].Quality < 50)
-                                {
-                                    Items[i].Quality = Items[i].Quality + 1;
-                                }
-                            }
-
-                            if (Items[i].SellIn < 6)
-                            {
-                                if (Items[i].Quality < 50)
-                                {
-                                    Items[i].Quality = Items[i].Quality + 1;
-                                }
-                            }
-                        }
-                    }
+                    VerifyLegendaryItem(item);
+                    //"Sulfuras", being a legendary item, never has to be sold or decreases in Quality
+                    break;
                 }
 
-                if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
-                {
-                    Items[i].SellIn = Items[i].SellIn - 1;
-                }
+                var qualityChange = CalculateQualityChange(item);
+                item.SellIn--;
+                item.Quality += qualityChange;
+                item.Quality = QualityOverflowHandler(item);
+            }
+        }
 
-                if (Items[i].SellIn < 0)
-                {
-                    if (Items[i].Name != "Aged Brie")
-                    {
-                        if (Items[i].Name != "Backstage passes to a TAFKAL80ETC concert")
-                        {
-                            if (Items[i].Quality > 0)
-                            {
-                                if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
-                                {
-                                    Items[i].Quality = Items[i].Quality - 1;
-                                }
-                            }
-                        }
-                        else
-                        {
-                            Items[i].Quality = Items[i].Quality - Items[i].Quality;
-                        }
-                    }
-                    else
-                    {
-                        if (Items[i].Quality < 50)
-                        {
-                            Items[i].Quality = Items[i].Quality + 1;
-                        }
-                    }
-                }
+        private int CalculateQualityChange(Item item)
+        {
+            if (item.Name.Contains("Backstage passes"))
```

Comment:
You can assign whole string "Backstage passes to a TAFKAL80ETC concert" to a constant and check if Name is equal to this constant. 

You can also try Enum as well